src/DockContainerWidget.cpp

```diff
@@ -1189,6 +1189,12 @@ bool DockContainerWidgetPrivate::restoreDockArea(CDockingStateReader& s,
 			continue;
 		}
 
+		const auto oldDockArea = DockWidget->dockAreaWidget();

```

Comment:
Could you please provide a small test case and instructions how to generate a crash here. I use the library since years in our product and have never seen any crash here so I would like to investigate the reason for this crash.

Reply:
On further investigation, it is related to the auto hide feature. When restoring the [auto hide container](https://github.com/duerr-ndt/Qt-Advanced-Docking-System/blob/d56281e8600d3a9c9648afa7deb6674ae4654169/src/DockContainerWidget.cpp#L1118) the function `CAutoHideDockContainer::addDockWidget` internally [removes the old dock area](https://github.com/duerr-ndt/Qt-Advanced-Docking-System/blob/d56281e8600d3a9c9648afa7deb6674ae4654169/src/AutoHideDockContainer.cpp#L291). This is what triggers the crash. I guess you would never see the crash if you don't remove dock widgets while restoring (which was not done previously).

In this case, I'll preserve the old behavior, and simply not remove the dock widget in the autohide container when restoring in order to keep the behavior the same :)

Reply:
Unfortunately, I can't get it to crash except with our application, but I will post a test case if I manage to.